openmc/mgxs/library.py

```diff
@@ -717,3 +725,307 @@ def load_from_file(filename='mgxs', directory='mgxs'):
 
         # Load and return pickled Library object
         return pickle.load(open(full_filename, 'rb'))
+
+    def write_mg_library(self, xs_type='macro', domain_names=None, xs_ids=None,
+                         filename='mg_cross_sections', directory='./',
+                         return_names=True):
+        """Creates a cross-section data library file for the Multi-Group
+        mode of OpenMC.
+
+        Parameters
+        ----------
+        xs_type: {'macro', 'micro'}
+            Provide the macro or micro cross section in units of cm^-1 or
+            barns. Defaults to 'macro'. If the Library object is not tallied by
+            nuclide this will be set to 'macro' regardless.
+        domain_names : Iterable of str
+            List of names to apply to the xsdata entries in the
+            resultant mgxs data file. Defaults to "set1", "set2", ...
+        xs_ids : str or Iterable of str
+            Cross section set identifier (i.e., "71c") for all
+            data sets (if only str) or for each individual one
+            (if iterable of str). Defaults to '1g'
```

Comment:
well in this case, the `1` does not refer to the number of groups.  Its just an arbitrary number that the user can have be whatever they wish.  I think that's reasonable as if one has data for the same nuclide with the same number of groups but from different temperatures or source data (i.e., endf revs), then the number being the number of groups isn't meaningful anymore.  So the `1` is simply a number and nothing more.  I think my letter flag, `g`, led you in to thinking that, and that's why now I'll switch to `m`.